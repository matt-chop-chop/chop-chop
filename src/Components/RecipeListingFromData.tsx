import { useRecipe } from "@/hooks";
import { ErrorCard, LoadingCard, RecipeListing } from "@/Components";

type RecipeListingProps = {
  id: string;
};

const RecipeListingFromData = ({ id }: RecipeListingProps) => {
  const { recipe, loading, error } = useRecipe(id);

  if (loading) {
    return <LoadingCard />;
  }

  if (error) {
    return <ErrorCard error={error} />;
  }

  if (!recipe) {
    return <ErrorCard error={new Error("Recipe could not be found")} />;
  }

  return <RecipeListing recipe={recipe} />;
};

export default RecipeListingFromData;
