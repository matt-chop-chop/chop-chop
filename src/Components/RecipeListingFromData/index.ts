export { default as RecipeListingFromData } from "./RecipeListingFromData";
