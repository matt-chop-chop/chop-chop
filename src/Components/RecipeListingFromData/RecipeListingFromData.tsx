import { useRecipe } from "@/hooks";
import { ErrorCard, LoadingCard, RecipeListing } from "@/Components";
import React from "react";

export type RecipeListingFromDataProps = {
  id: string;
};

const RecipeListingFromData = ({ id }: RecipeListingFromDataProps) => {
  const { recipe, loading, error } = useRecipe(id);

  if (loading) {
    return <LoadingCard />;
  }

  if (error) {
    return <ErrorCard error={error} />;
  }

  if (!recipe) {
    return <ErrorCard error={new Error("Recipe could not be found")} />;
  }

  return <RecipeListing recipe={recipe} />;
};

export default RecipeListingFromData;
