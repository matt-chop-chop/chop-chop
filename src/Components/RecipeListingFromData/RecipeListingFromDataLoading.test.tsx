import { shallow } from "enzyme";
import React from "react";
import RecipeListingFromData, {
  RecipeListingFromDataProps,
} from "./RecipeListingFromData";

jest.mock("@/hooks/useRecipe", () => ({
  __esModule: true,
  useRecipe: () => ({ recipe: null, loading: true }),
}));

const props: RecipeListingFromDataProps = {
  id: "123",
};

describe("RecipeListingFromData", () => {
  it("is exported from RecipeListingFromData", () => {
    expect(RecipeListingFromData).toBeDefined();
  });

  it("renders as expected when loading", () => {
    const wrapper = shallow(<RecipeListingFromData {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
