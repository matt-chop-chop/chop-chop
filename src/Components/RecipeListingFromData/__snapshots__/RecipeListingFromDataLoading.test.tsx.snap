// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeListingFromData renders as expected when loading 1`] = `<LoadingCard />`;
